v2client/common/models/v2models.go

```diff
@@ -0,0 +1,1009 @@
+	GenesisHash []byte       `json:"genesis-hash"`
+	GenesisId   string       `json:"genesis-id"`
+	Versions    []string     `json:"versions"`
+}
+
+// VersionBuild defines model for the current algod build version information.
+type VersionBuild struct {
+	Branch      string `json:"branch"`
+	BuildNumber uint64 `json:"build-number"`
+	Channel     string `json:"channel"`
+	CommitHash  []byte `json:"commit-hash"`
+	Major       uint64 `json:"major"`
+	Minor       uint64 `json:"minor"`
+}
+
+// AccountId defines model for account-id.
+type AccountId string
+
+// Address defines model for address.
+type Address string
+
+// AddressGreaterThan defines model for address-greater-than.
+type AddressGreaterThan string
+
+// AddressRole defines model for address-role.
+type AddressRole string
+
+// AfterAddress defines model for after-address.
+type AfterAddress string
+
+// AfterAsset defines model for after-asset.
+type AfterAsset uint64
+
+// AfterTime defines model for after-time.
+type AfterTime time.Time
+
+// AlgosGreaterThan defines model for algos-greater-than.
+type AlgosGreaterThan uint64
+
+// AlgosLessThan defines model for algos-less-than.
+type AlgosLessThan uint64
+
+// AssetId defines model for asset-id.
+type AssetId uint64
+
+// BeforeTime defines model for before-time.
+type BeforeTime time.Time
+
+// CurrencyGreaterThan defines model for currency-greater-than.
+type CurrencyGreaterThan uint64
+
+// CurrencyLessThan defines model for currency-less-than.
+type CurrencyLessThan uint64
+
+// ExcludeCloseTo defines model for exclude-close-to.
+type ExcludeCloseTo bool
+
+// Limit defines model for limit.
+type Limit uint64
+
+// MaxRound defines model for max-round.
+type MaxRound uint64
+
+// MinRound defines model for min-round.
+type MinRound uint64
+
+// NotePrefix defines model for note-prefix.
+type NotePrefix []byte
+
+// Offset defines model for offset.
+type Offset uint64
+
+// Round defines model for round.
+type Round uint64
+
+// RoundNumber defines model for round-number.
+type RoundNumber uint64
+
+// SigType defines model for sig-type.
+type SigType string
+
+// TxId defines model for tx-id.
+type TxId []byte
+
+// TxType defines model for tx-type.
+type TxType string
+
+// AccountResponse defines model for AccountResponse.
+type AccountResponse struct {
+
+	// Account information at a given round.
+	//
+	// Definition:
+	// data/basics/userBalance.go : AccountData
+	Account Account `json:"account"`
+
+	// Round at which the results were computed.
+	CurrentRound uint64 `json:"current-round"`
+}
+
+// AccountsResponse defines model for AccountsResponse.
+type AccountsResponse struct {
+	Accounts []Account `json:"accounts"`
+
+	// Round at which the results were computed.
+	CurrentRound uint64 `json:"current-round"`
+}
+
+// AssetBalancesResponse defines model for AssetBalancesResponse.
+type AssetBalancesResponse struct {
+
+	// A simplified version of AssetHolding
+	Balances []MiniAssetHolding `json:"balances"`
```

Comment:
@winder the spec specifies this as a single MiniAssetHolding but it looks like it should come back as an array, so I changed that.